cmd/charmd/main.go

```diff
@@ -40,25 +40,32 @@ func main() {
 	if flag.NArg() != 1 {
 		flag.Usage()
 	}
+	fmt.Fprintf(os.Stderr, "START\n")
 	if *loggingConfig != "" {
 		if err := loggo.ConfigureLoggers(*loggingConfig); err != nil {
-			fmt.Fprintf(os.Stderr, "cannot configure loggers: %v", err)
-			os.Exit(1)
+			fmt.Fprintf(os.Stderr, "STOP: cannot configure loggers: %s\n", err)
+			os.Exit(2)
 		}
 	}
-	if err := serve(flag.Arg(0)); err != nil {
-		fmt.Fprintf(os.Stderr, "%v\n", err)
-		os.Exit(1)
-	}
-}
-
-func serve(confPath string) error {
 	logger.Infof("reading configuration")
-	conf, err := config.Read(confPath)
+	conf, err := config.Read(flag.Arg(0))
 	if err != nil {
-		return errgo.Notef(err, "cannot read config file %q", confPath)
+		fmt.Fprintf(os.Stderr, "STOP: %s\n", err)
+		os.Exit(2)
 	}
+	if *loggingConfig == "" && conf.LoggingConfig != "" {
+		if err := loggo.ConfigureLoggers(*loggingConfig); err != nil {
```

Comment:
This looks wrong to me. I think you want conf.LoggingConfig here, otherwise this line is guaranteed to do nothing.